### PR TITLE
Weekly cumulative set of changes

### PR DESCRIPTION
**Pull request by pzaino**
This PR contains:
- [ ] bugfix
- [ ] feature
- [ ] refactor
- [x] documentation
- [ ] other

Are tests included?
- [x] no
- [ ] yes (*bugfixes and features will not be merged without tests*)

Breaking Changes?
- [x] no
- [ ] yes (*breaking changes will not be merged unless absolutely necessary – please provide motivation*)

List any relevant issue numbers:

### Description

This PR contains a cumulative set of changes (some weekly automated ones and some added as required by the original Lua license). 

<b>Please note:</b> if you find issues in CODE_OF_CONDUCT.MD and/or CONTRIBUTING.MD they will need to be fixed in this repository: https://github.com/RISC-OS-Community/StandardRepoTemplate 

thanks :)

### CODE REVIEW

**Comment by GavinWraith** [comment]
Paolo

On 2021-11-05 18:37, Paolo Fabio Zaino wrote:
> @pzaino [1] requested your review on: #4 [2] Weekly cumulative set of
> changes.

How do I edit what I see in the github pages? Years of using RISC OS 
where reading a text file means using a text-editor has accustomed me to 
being able to edit anything I am reading just by clicking the mouse at 
the point I want to insert or delete and typing away.

I think I saw 'accoutn' instead of 'account' in one of the files, and 
the README.md needs a line

     package.cpath: rlua86:solib.?/so,?/so

after the line for package.path.

I was going to ask you some general Linux advice. The Rpi4 I use for 
Linux is set to boot first from the micro-SD card, if it is inserted, 
and if not to boot from the first device connected by USB3. Normally I 
have no micro-SD card inserted, and it boots from what I have connected. 
In the case of Raspbian that is a Manx Data SSD drive. /dev/sda1 is the 
boot partition and /dev/sda2 is Linux (much bigger, about 245 Gb). Just 
recently I have been getting errors at boot up, or odd things have 
happened: e.g. trying to open the Chrome Browser makes the apps-bar 
disappear. I then
switch off and things are usually OK when I switch on again. I did
    sudo touch /forcefsck
but could not see anything in /var/log/daemon.log . Then I did

    sudo badblocks -v /dev/sda2 >badsectors.txt

and after a very long time it reported no bad blocks.

I am going to do the same with /dev/sda1 .

Basically I am just following whatever advice I can find on the 
internet. Maybe you have some wise words on this topic?

-- Gavin

**Comment by pzaino** [comment]
@GavinWraith ,

>the README.md needs a line
>
>     package.cpath: rlua86:solib.?/so,?/so
>
> after the line for package.path.

Ok, added, thanks for the info. :)

Generally in a review you ask the person who has requested the pull to change what you've found to be incorrect. It is also possible to propose the change yourself by doing it in the review process you enable by clicking "do your review" button (or similar) on the top of the page. But you can learn how to use that process later, no worries.

> I think I saw 'accoutn' instead of 'account' in one of the files

I think I see it in one of the two special files. As explained previously, issues in CODE_OF_CONDUCT.MD and CONTRIBUTING.MD needs to be done in a different repository, which is pulled it weekly to all our other repositories, so we do not need to do the sam change everywhere :)

I'll fix it in this repo: https://github.com/RISC-OS-Community/StandardRepoTemplate and then will do a new pull request for the change.

On the Linux issues, we should take this discussion off-line, but as a quick help:

> In the case of Raspbian that is a Manx Data SSD drive. /dev/sda1 is the
> boot partition and /dev/sda2 is Linux (much bigger, about 245 Gb)

Yup, this is normal.

> I am going to do the same with /dev/sda1 .

Yes, so Linux is a bit of an atypical OS. when it boots, it boots the kernel and a special File System image that are bot contained in the boot partition. When the kernel is fully booted then yes it mounts the 2nd partition and start booting stuff from your /dev/sda2 partition. So yes you want to check also sda1 for bad sectors.

It's important to note that errors during boot could be caused by few things:

1) Kernel and RPi firmware not working well together, this  usually can cause bigger problems that the one you are seeing, so I don't think this is the case. However this can be solved updating the firmware to the next STABLE one by using an alternative boot, for example from a USB key.

2) SD corruption, definitely not your case. This can be solved (to a certain extent) by using specific tools for the FileSystem you are using on a corrupted partition.

3) Issues with the specific release of the OS you have, this could be solved with an update if available.

If you need, I can prepare for you a clean Installation on an SD card of Raspian or RaspberryPi OS so you have a safe bootable, spare one. Let me know via email please.

Thanks
